src/set_theory/game/pgame.lean

```diff
@@ -261,287 +249,284 @@ instance : has_one pgame := ⟨⟨punit, pempty, λ _, 0, pempty.elim⟩⟩
-classical.some $ (le_zero.1 h) i
+instance : preorder pgame :=
+{ le_refl := λ x, begin
+    induction x with _ _ _ _ IHl IHr,
+    exact le_def_lf.2 ⟨λ i, lf_of_le_move_left (IHl i), λ i, lf_of_move_right_le (IHr i)⟩
+  end,
+  le_trans := λ x y z, suffices ∀ {x y z : pgame},
+    (x ≤ y → y ≤ z → x ≤ z) ∧ (y ≤ z → z ≤ x → y ≤ x) ∧ (z ≤ x → x ≤ y → z ≤ y),
+  from this.1, begin
+    clear x y z, intros,
+    induction x with xl xr xL xR IHxl IHxr generalizing y z,
+    induction y with yl yr yL yR IHyl IHyr generalizing z,
+    induction z with zl zr zL zR IHzl IHzr,
+    exact ⟨
+      le_trans_aux (λ i, (IHxl _).2.1) (λ i, (IHzr _).2.2),
+      le_trans_aux (λ i, (IHyl _).2.2) (λ i, (IHxr _).1),
+      le_trans_aux (λ i, (IHzl _).1) (λ i, (IHyr _).2.1)⟩,
+  end,
+  ..pgame.has_le }
 
-/-- Show that the response for right provided by `right_response`
-    preserves the right-player-wins condition. -/
-lemma right_response_spec {x : pgame} (h : x ≤ 0) (i : x.left_moves) :
-  (x.move_left i).move_right (right_response h i) ≤ 0 :=
-classical.some_spec $ (le_zero.1 h) i
+theorem lf_irrefl (x : pgame) : ¬ x ⧏ x := pgame.not_lf.2 le_rfl
+instance : is_irrefl _ (⧏) := ⟨lf_irrefl⟩
 
-/-- Given a left-player-wins game, provide a response to any move by right. -/
-noncomputable def left_response {x : pgame} (h : 0 ≤ x) (j : x.right_moves) :
-  (x.move_right j).left_moves :=
-classical.some $ (zero_le.1 h) j
+@[trans] theorem lf_of_le_of_lf {x y z : pgame} (h₁ : x ≤ y) (h₂ : y ⧏ z) : x ⧏ z :=
+by { rw ←pgame.not_le at h₂ ⊢, exact λ h₃, h₂ (h₃.trans h₁) }
 
-/-- Show that the response for left provided by `left_response`
-    preserves the left-player-wins condition. -/
-lemma left_response_spec {x : pgame} (h : 0 ≤ x) (j : x.right_moves) :
-  0 ≤ (x.move_right j).move_left (left_response h j) :=
-classical.some_spec $ (zero_le.1 h) j
+@[trans] theorem lf_of_lf_of_le {x y z : pgame} (h₁ : x ⧏ y) (h₂ : y ≤ z) : x ⧏ z :=
+by { rw ←pgame.not_le at h₁ ⊢, exact λ h₃, h₁ (h₂.trans h₃) }
+
+theorem move_left_lf {x : pgame} (i) : x.move_left i ⧏ x :=
+move_left_lf_of_le i le_rfl
 
-theorem lt_of_le_mk {xl xr xL xR y i} :
-  (⟨xl, xr, xL, xR⟩ : pgame) ≤ y → xL i < y :=
-by { cases y, rw mk_le_mk, tauto }
+theorem lf_move_right {x : pgame} (j) : x ⧏ x.move_right j :=
+lf_move_right_of_le j le_rfl
 
-theorem lt_of_mk_le {x : pgame} {yl yr yL yR i} :
-  x ≤ ⟨yl, yr, yL, yR⟩ → x < yR i :=
-by { cases x, rw mk_le_mk, tauto }
+theorem lt_iff_le_and_lf {x y : pgame} : x < y ↔ x ≤ y ∧ x ⧏ y :=
+by rw [lt_iff_le_not_le, pgame.not_le]
 
-theorem mk_lt_of_le {xl xr xL xR y i} :
-  ((xR : xr → pgame) i ≤ y) → (⟨xl, xr, xL, xR⟩ : pgame) < y :=
-by { cases y, rw mk_lt_mk, tauto }
+theorem lt_of_le_of_lf {x y : pgame} (h₁ : x ≤ y) (h₂ : x ⧏ y) : x < y :=
+lt_iff_le_and_lf.2 ⟨h₁, h₂⟩
 
-theorem lt_mk_of_le {x : pgame} {yl yr : Type*} {yL : yl → pgame} {yR i} :
-  (x ≤ yL i) → x < ⟨yl, yr, yL, yR⟩ :=
-by { cases x, rw mk_lt_mk, exact λ h, or.inl ⟨_, h⟩ }
```

Comment:
Seems these are also removed (in favor of the move_left/move_right versions)?

Reply:
Yeah, same reasoning as above.